# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1497970497 14400
#      Tue Jun 20 10:54:57 2017 -0400
# Node ID 609658f98feb937bf61e6a435c62f0ea5b63c6ba
# Parent  bbcc1e08aaed9d4ce7b6879a42264d0406a92b5c
remove some unnecessary include statements from parse tree source
files

* pt-array-list.cc, pt-assign.cc, pt-binop.cc, pt-cbinop.cc,
pt-fcn-handle.cc, pt-id.cc, pt-idx.cc, pt-jump.cc, pt-loop.cc,
pt-mat.cc, pt-misc.cc, pt-select.cc, pt-stmt.cc, pt-unop.cc, token.cc:
Remove some unnecessary include statements.

diff --git a/libinterp/parse-tree/pt-array-list.cc b/libinterp/parse-tree/pt-array-list.cc
--- a/libinterp/parse-tree/pt-array-list.cc
+++ b/libinterp/parse-tree/pt-array-list.cc
@@ -21,16 +21,18 @@ along with Octave; see the file COPYING.
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include <iostream>
 
+#include "quit.h"
+
 #include "error.h"
 #include "pt-array-list.h"
 
 namespace octave
 {
   tree_array_list::~tree_array_list (void)
   {
     while (! empty ())
diff --git a/libinterp/parse-tree/pt-assign.cc b/libinterp/parse-tree/pt-assign.cc
--- a/libinterp/parse-tree/pt-assign.cc
+++ b/libinterp/parse-tree/pt-assign.cc
@@ -20,33 +20,22 @@ along with Octave; see the file COPYING.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include <iostream>
-#include <set>
+#include <string>
 
-#include "defun.h"
 #include "error.h"
-#include "errwarn.h"
-#include "input.h"
-#include "ovl.h"
-#include "oct-lvalue.h"
 #include "ov.h"
-#include "pager.h"
-#include "parse.h"
 #include "pt-arg-list.h"
-#include "pt-bp.h"
 #include "pt-assign.h"
-#include "pt-walk.h"
-#include "utils.h"
-#include "variables.h"
 
 namespace octave
 {
   // Simple assignment expressions.
 
   tree_simple_assignment::tree_simple_assignment
   (tree_expression *le, tree_expression *re,
    bool plhs, int l, int c, octave_value::assign_op t)
diff --git a/libinterp/parse-tree/pt-binop.cc b/libinterp/parse-tree/pt-binop.cc
--- a/libinterp/parse-tree/pt-binop.cc
+++ b/libinterp/parse-tree/pt-binop.cc
@@ -20,23 +20,18 @@ along with Octave; see the file COPYING.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include "error.h"
-#include "defun.h"
-#include "ovl.h"
 #include "ov.h"
-#include "profiler.h"
 #include "pt-binop.h"
-#include "pt-bp.h"
-#include "pt-walk.h"
 #include "variables.h"
 
 namespace octave
 {
   // Binary expressions.
 
   void
   tree_binary_expression::matlab_style_short_circuit_warning (const char *op)
diff --git a/libinterp/parse-tree/pt-cbinop.cc b/libinterp/parse-tree/pt-cbinop.cc
--- a/libinterp/parse-tree/pt-cbinop.cc
+++ b/libinterp/parse-tree/pt-cbinop.cc
@@ -19,23 +19,19 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include "error.h"
-#include "ovl.h"
 #include "ov.h"
 #include "pt-cbinop.h"
-#include "pt-bp.h"
 #include "pt-unop.h"
-#include "pt-walk.h"
 
 namespace octave
 {
   typedef tree_expression* tree_expression_ptr_t;
 }
 
 // If a tree expression is a transpose or hermitian transpose, return
 // the argument and corresponding operator.
diff --git a/libinterp/parse-tree/pt-fcn-handle.cc b/libinterp/parse-tree/pt-fcn-handle.cc
--- a/libinterp/parse-tree/pt-fcn-handle.cc
+++ b/libinterp/parse-tree/pt-fcn-handle.cc
@@ -21,26 +21,18 @@ along with Octave; see the file COPYING.
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include <iostream>
 
-#include "call-stack.h"
-#include "error.h"
 #include "interpreter-private.h"
-#include "ovl.h"
-#include "ov-fcn-handle.h"
 #include "pt-fcn-handle.h"
-#include "pager.h"
-#include "pt-const.h"
-#include "pt-walk.h"
-#include "variables.h"
 
 namespace octave
 {
   void
   tree_fcn_handle::print (std::ostream& os, bool pr_as_read_syntax,
                           bool pr_orig_text)
   {
     print_raw (os, pr_as_read_syntax, pr_orig_text);
diff --git a/libinterp/parse-tree/pt-id.cc b/libinterp/parse-tree/pt-id.cc
--- a/libinterp/parse-tree/pt-id.cc
+++ b/libinterp/parse-tree/pt-id.cc
@@ -21,33 +21,30 @@ along with Octave; see the file COPYING.
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include "error.h"
 #include "interpreter-private.h"
-#include "ovl.h"
 #include "oct-lvalue.h"
-#include "pager.h"
 #include "parse.h"
-#include "pt-bp.h"
 #include "pt-const.h"
-#include "pt-eval.h"
 #include "pt-id.h"
-#include "pt-walk.h"
 #include "symtab.h"
 #include "utils.h"
 #include "variables.h"
 
 namespace octave
 {
   // Symbols from the symbol table.
 
+  class tree_evaluator;
+
   void
   tree_identifier::eval_undefined_error (void)
   {
     int l = line ();
     int c = column ();
 
     maybe_missing_function_hook (name ());
 
diff --git a/libinterp/parse-tree/pt-idx.cc b/libinterp/parse-tree/pt-idx.cc
--- a/libinterp/parse-tree/pt-idx.cc
+++ b/libinterp/parse-tree/pt-idx.cc
@@ -26,23 +26,20 @@ along with Octave; see the file COPYING.
 
 #include "Cell.h"
 #include "error.h"
 #include "interpreter-private.h"
 #include "oct-map.h"
 #include "ovl.h"
 #include "oct-lvalue.h"
 #include "ov.h"
-#include "pager.h"
 #include "pt-arg-list.h"
-#include "pt-bp.h"
 #include "pt-eval.h"
 #include "pt-id.h"
 #include "pt-idx.h"
-#include "pt-walk.h"
 #include "utils.h"
 #include "variables.h"
 #include "errwarn.h"
 
 namespace octave
 {
   // Index expressions.
 
diff --git a/libinterp/parse-tree/pt-jump.cc b/libinterp/parse-tree/pt-jump.cc
--- a/libinterp/parse-tree/pt-jump.cc
+++ b/libinterp/parse-tree/pt-jump.cc
@@ -19,21 +19,17 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include "error.h"
-#include "ovl.h"
-#include "pt-bp.h"
 #include "pt-jump.h"
-#include "pt-walk.h"
 
 class octave_value_list;
 
 namespace octave
 {
   // Break.
 
   // Nonzero means we're breaking out of a loop or function body.
diff --git a/libinterp/parse-tree/pt-loop.cc b/libinterp/parse-tree/pt-loop.cc
--- a/libinterp/parse-tree/pt-loop.cc
+++ b/libinterp/parse-tree/pt-loop.cc
@@ -19,33 +19,19 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include "quit.h"
-
-#include "error.h"
-#include "errwarn.h"
-#include "oct-map.h"
-#include "oct-lvalue.h"
-#include "ov.h"
 #include "pt-arg-list.h"
-#include "pt-bp.h"
-#include "pt-cmd.h"
-#include "pt-exp.h"
-#include "pt-jit.h"
-#include "pt-jump.h"
 #include "pt-loop.h"
 #include "pt-stmt.h"
-#include "pt-walk.h"
-#include "unwind-prot.h"
 
 namespace octave
 {
   // While.
 
   tree_while_command::~tree_while_command (void)
   {
     delete expr;
diff --git a/libinterp/parse-tree/pt-mat.cc b/libinterp/parse-tree/pt-mat.cc
--- a/libinterp/parse-tree/pt-mat.cc
+++ b/libinterp/parse-tree/pt-mat.cc
@@ -19,34 +19,22 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <iostream>
-
-#include "oct-locbuf.h"
-#include "quit.h"
-
-#include "data.h"
 #include "defun.h"
-#include "error.h"
-#include "errwarn.h"
-#include "oct-map.h"
 #include "ovl.h"
 #include "pt-arg-list.h"
-#include "pt-bp.h"
 #include "pt-exp.h"
 #include "pt-mat.h"
 #include "pt-tm-const.h"
-#include "pt-walk.h"
-#include "utils.h"
 #include "ov.h"
 #include "variables.h"
 
 #include "ov-cx-mat.h"
 #include "ov-flt-cx-mat.h"
 #include "ov-re-sparse.h"
 #include "ov-cx-sparse.h"
 
diff --git a/libinterp/parse-tree/pt-misc.cc b/libinterp/parse-tree/pt-misc.cc
--- a/libinterp/parse-tree/pt-misc.cc
+++ b/libinterp/parse-tree/pt-misc.cc
@@ -19,27 +19,18 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include "Cell.h"
-
-#include "defun.h"
-#include "error.h"
-#include "ov.h"
-#include "oct-lvalue.h"
-#include "pt-id.h"
 #include "pt-idx.h"
 #include "pt-misc.h"
-#include "pt-walk.h"
-#include "utils.h"
 
 namespace octave
 {
   // Parameter lists.
 
   tree_parameter_list::~tree_parameter_list (void)
   {
     while (! empty ())
diff --git a/libinterp/parse-tree/pt-select.cc b/libinterp/parse-tree/pt-select.cc
--- a/libinterp/parse-tree/pt-select.cc
+++ b/libinterp/parse-tree/pt-select.cc
@@ -19,26 +19,18 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include "error.h"
-#include "ovl.h"
-#include "ov.h"
-#include "pt-cmd.h"
-#include "pt-exp.h"
 #include "pt-select.h"
 #include "pt-stmt.h"
-#include "pt-walk.h"
-#include "Cell.h"
-#include "ov-typeinfo.h"
 
 namespace octave
 {
   // If clauses.
 
   tree_if_clause::~tree_if_clause (void)
   {
     delete expr;
diff --git a/libinterp/parse-tree/pt-stmt.cc b/libinterp/parse-tree/pt-stmt.cc
--- a/libinterp/parse-tree/pt-stmt.cc
+++ b/libinterp/parse-tree/pt-stmt.cc
@@ -24,33 +24,29 @@ along with Octave; see the file COPYING.
 #  include "config.h"
 #endif
 
 #include <typeinfo>
 
 #include "quit.h"
 
 #include "bp-table.h"
-#include "defun.h"
-#include "error.h"
-#include "errwarn.h"
 #include "input.h"
 #include "oct-lvalue.h"
 #include "octave-link.h"
 #include "ov.h"
 #include "pager.h"
 #include "pt-bp.h"
 #include "pt-cmd.h"
 #include "pt-id.h"
 #include "pt-idx.h"
 #include "pt-jump.h"
 #include "pt-pr-code.h"
 #include "pt-stmt.h"
 #include "pt-walk.h"
-#include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 
 namespace octave
 {
   // A list of commands to be executed.
 
   tree_statement::~tree_statement (void)
diff --git a/libinterp/parse-tree/pt-unop.cc b/libinterp/parse-tree/pt-unop.cc
--- a/libinterp/parse-tree/pt-unop.cc
+++ b/libinterp/parse-tree/pt-unop.cc
@@ -19,24 +19,18 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include "error.h"
-#include "ovl.h"
-#include "oct-lvalue.h"
 #include "ov.h"
-#include "profiler.h"
-#include "pt-bp.h"
 #include "pt-unop.h"
-#include "pt-walk.h"
 
 namespace octave
 {
   // Unary expressions.
 
   std::string
   tree_unary_expression::oper (void) const
   {
diff --git a/libinterp/parse-tree/token.cc b/libinterp/parse-tree/token.cc
--- a/libinterp/parse-tree/token.cc
+++ b/libinterp/parse-tree/token.cc
@@ -21,21 +21,18 @@ along with Octave; see the file COPYING.
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include <cassert>
 
-#include "error.h"
-#include "ovl.h"
 #include "symtab.h"
 #include "token.h"
-#include "utils.h"
 
 namespace octave
 {
   token::token (int tv, int l, int c)
     : m_maybe_cmd (false), m_tspc (false), m_line_num (l), m_column_num (c),
       m_tok_val (tv), m_type_tag (generic_token), m_tok_info (),
       m_orig_text ()
   { }
